### ISSUE TITLE
EF .net core discriminator inheritance leading to wrong metadata structure

### ISSUE THREAD

**Issue by graphicsxp** [Breeze/breeze.server.net#99]
We are having an issue with the following mapping. The resulting JSON metadata seem to be wrong. 

Basically `PrintRequest` inherits from `Request`  so in the metadata we are not expecting to see duplicated properties such as `Id` in the dataProperties array of both classes, but only in the dataProperties of the base class, in this case `Request` 

Is this the correct behavior ?  

 **EF Configuration**

        public partial class RequestConfiguration : BusinessEntityConfiguration<Request>
    {
        public override void Configure(EntityTypeBuilder<Request> builder)
        {
            base.Configure(builder);
            builder
                .ToTable("Request", "author")
                .HasDiscriminator(t => t.Type)
                .HasValue<PrintRequest>(typeof(PrintRequest).FullName);
            builder.HasKey(e => e.Id).IsClustered(false);
            builder.Property(e => e.Id).ValueGeneratedOnAdd();
         }
     }

**Models**
```
   public abstract class Entity : IEntity
    {
        public Guid Id { get; set; } 
    }
    public abstract class BusinessEntity : Entity
    {
        public Guid StateId { get; set; }
    }

    public class Request : BusinessEntity
    {
        public string Type { get; set; }
     }

    public partial class PrintRequest : Request
    {
    }

```

**Metadata excerpt** 

```
"structuralTypes": [
        {
            "shortName": "PrintRequest",
            "namespace": "Op.Authors",
            "baseTypeName": "Request:#Op.Authors",
            "autoGeneratedKeyType": "Identity",
            "defaultResourceName": "PrintRequests",
            "isComplexType": false,
            "dataProperties": [
                {
                    "dataType": "Guid",
                    "isPartOfKey": true,
                    "isNullable": false,
                    "nameOnServer": "Id",
                    "validators": [
                        {
                            "name": "required"
                        },
                        {
                            "name": "guid"
                        }
                    ]
                },

.....

{
            "shortName": "Request",
            "namespace": "Op.Authors",
            "autoGeneratedKeyType": "Identity",
            "isComplexType": false,
            "dataProperties": [
                {
                    "dataType": "Guid",
                    "isPartOfKey": true,
                    "isNullable": false,
                    "nameOnServer": "Id",
                    "validators": [
                        {
                            "name": "required"
                        },
                        {
                            "name": "guid"
                        }
                    ]
                },
```

**Comment by graphicsxp** [comment]
In breeze.persistence.EF.Core MetadataBuilder function there is the following line of code :
` mt.DataProperties = et.GetProperties().Select(p => CreateDataProperty(p)).ToList();`

If `et` is a child class, `et.GetProperties()` will return all properties, including the parent's ones. This is how we end up with the dataProperties shown in the JSON structure above. 

Is this the expected behavior ?  We believe it is not, properties should not be duplicated in child classes. This causes other errors when breeze build entities on the client side (but I'm not going into this yet until I know if we are doing the right thing server-side).

**Comment by graphicsxp** [comment]
any chance my pull request could be approved regarding this issue  ? Is the breeze project still alive ?

**Comment by steveschmitt** [comment]
I think this is the correct behavior.  In the metadata, the inherited properties are repeated in the subclass so that the breeze client does not have to walk the inheritance tree every time a property value is changed.

**Comment by steveschmitt** [comment]
Fixed in version 5.0.4 for .NET 5
Fixed in version 3.1.4 for .NET 3.1
